magefile.go

```diff
@@ -0,0 +1,71 @@
+// Build a script to format and run tests of a Terraform module project
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+// The default target when the command executes `mage` in Cloud Shell
+var Default = RunAllTargets
+
+func main() {
+	Default()
+}
+
+// A build step that runs Clean, Format, Unit and Integration in sequence
+func RunAllTargets() {
+	mg.Deps(RunUnitTests)
+	mg.Deps(RunIntegrationTests)
+}
+
+// A build step that runs unit tests
+func RunUnitTests() error {
+	mg.Deps(CleanAll)
+	mg.Deps(FormatGolangFiles)
+	fmt.Println("INFO: Running unit tests...")
+	return sh.RunV("go", "test", "./...", "-run", "TestUT", "-v")
+}
+
+// A build step that runs integration tests
+func RunIntegrationTests() error {
+	mg.Deps(CleanAll)
+	mg.Deps(FormatGolangFiles)
+	fmt.Println("INFO: Running integration tests...")
+	return sh.RunV("go", "test", "./...", "-run", "TestIT", "-v", "-timeout", "99999s")
+}
+
+// A build step that formats both Terraform code and Go code
+func FormatGolangFiles() error {
```

Comment:
[question for the future] Should this be caught by linting and cause a build to fail? I think you may be able to run something like this to check if the terraform is correctly formatted:

```
# https://www.terraform.io/docs/commands/fmt.html
terraform fmt -check=true
```

Reply:
Agree that we should create a user story to support a proper TF Lint target within our test harness.